import_3dm/read3dm.py

```diff
@@ -0,0 +1,76 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTIBILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+import os.path
+import bpy
+import rhino3dm as r3d
+##import rhino3dm as r3d
+from .converters import *
+
+def read_3dm(context, filepath, import_hidden, import_hidden_layers):
+    
+    
+    top_collection_name = os.path.splitext(os.path.basename(filepath))[0]
+    if top_collection_name in context.blend_data.collections.keys():
+        toplayer = context.blend_data.collections[top_collection_name]
+    else:
+        toplayer = context.blend_data.collections.new(name=top_collection_name)
+
+    model = r3d.File3dm.Read(filepath)
+    
+    layerids = {}
+    materials = {}
+    
+    handle_materials(context, model, materials)
+    handle_layers(context, model, toplayer, layerids, materials,import_hidden_layers)
+        
+    for ob in model.Objects:
+        og=ob.Geometry
+        if og.ObjectType not in RHINO_TYPE_TO_IMPORT: continue
+        exporter = RHINO_TYPE_TO_IMPORT[og.ObjectType]
+        
+        attr = ob.Attributes
+        if not import_hidden and not attr.Visible: continue    
```

Comment:
Also nice to see this fixed, but please do this in a separate PR after this one has been completely handled.

Reply:
I created an issue for this that you can reference once you have a PR for it: #17